### ISSUE TITLE
Need to press spacebar twice

### ISSUE THREAD

**Issue by Sebbe909** [QL-Win/QuickLook#1483]
*Please search within existing issues before the submission. Has this issue already been reported?*
Not that I'm aware of

**Describe the bug**
Since some while now, for more than a year, you sometimes need to press/tap the spacebar twice before the preview window is opened up. Sometimes it opens up after one press. But sometimes it requires two presses. I couldn't find any relation to file format.

**To Reproduce**
1. Select file
2. Press spacebar
3. Sometimes nothing happens

**Expected behavior**
The preview window should show up after pressing spacebar once.

**Desktop (please complete the following information):**
 - OS Version: Windows 11 24H2
 - QuickLook Version: 3.7.3

**Comment by xupefei** [comment]
Maybe since https://github.com/QL-Win/QuickLook/commit/c10773249f43dee79b23be64da6eb74fcf66eab9?
Tldr: press and hold Space will show the preview window and close it when the key is released.
The trigger threshold is 0.75 seconds. Are you pressing the key for a time longer than that?

**Comment by Sebbe909** [comment]
I believe it happened more recently than 3.7.0. But I also never "long press" the space, so it can't be that. Now I think about it, I believe this began around the time Explorer got tabs. You've never experienced this yourself? I first thought it maybe was my older PC. But I've just got a brand new one, and the same issue exists.

Could it possibly be that I'm pressing space too quickly after selecting a file/folder? I tend to work quite fast. Also note that I have the details pane enabled in explorer. I haven't dug into the QL code, but maybe it could be related to QL needing to process stuff before it's able to detect an action?

Still weird though that the issue didn't occur in the past, so I'm just thinking out loud here.

**Comment by rabelux** [comment]
I'm also having this issue regularly on Win 10 and my impression so far is, that the first trigger after system startup or a long time not using QL does not open a preview window. To me it looks like some caches first need to be filled and then it works.

**Comment by Sebbe909** [comment]
@xupefei any chance to look into this or maybe have an update? :)

**Comment by mooflu** [comment]
> I'm also having this issue regularly on Win 10 and my impression so far is, that the first trigger after system startup or a long time not using QL does not open a preview window. To me it looks like some caches first need to be filled and then it works.

I'm on Win11 and I see this, too. Though maybe not the same as it does open a preview eventually after 5-6 secs. I.e. no need to press space again.

I can repro pretty reliably after computer comes out of sleep, the first preview takes 5-6sec after pressing space bar. During the 5-6secs the screen freezes (mouse can still move) and can't interact with anything. Which makes it a pain to diagnose...

Tried with an older release and happens there, too. 
Tried killing explorer and open QL preview via a dialog - same thing.
Tried disabling real-time protection temporarily - same thing.
Tried with network cable disconnected - same thing.
When running in debugger and recording cpu profile most cpu is in system calls (ntdll.dll). 

https://github.com/QL-Win/QuickLook/issues/1526 and https://github.com/QL-Win/QuickLook/issues/1521 sound similar, too.

**Comment by mooflu** [comment]
Ran with Windows Performance Recorder and looks like a bulk of the time is spent in the nvidia driver and look like it's stalling
```
...
nvlddmkm.sys!<PDB not found>
ntoskrnl.exe!KeStallExecutionProcessor
ntoskrnl.exe!HalpTimerStallExecutionProcessor
```

That's with driver version 560.94

Upgraded to 572.47 - no change.

Tried an older version from around Aug 2023 - 536.99 and I can no longer reproduce.

**Comment by emako** [comment]
Done in preview version.

The fix prevents the first Space keydown from being locked out before it is actually handled, so subsequent Space presses are no longer discarded.
